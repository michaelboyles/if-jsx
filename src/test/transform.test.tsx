import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Else, If } from '..';

const NOT_RENDERED = 'THIS TEXT SHOULD NOT BE RENDERED!';

it('Positive if', () => {
    const tree = renderer
        .create(<If condition={true}>Condition is true</If>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Positive if inside fragment', () => {
    const tree = renderer
        .create(<><If condition={true}>Condition is true</If></>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Positive if inside div', () => {
    const tree = renderer
        .create(<div><If condition={true}>Condition is true</If></div>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Positive if with evaluated condition', () => {
    let val = 3;
    const tree = renderer
        .create(<If condition={val === 3}>Condition is true</If>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Negative if inside div', () => {
    const tree = renderer
        .create(<div><If condition={false}>{NOT_RENDERED}</If></div>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Negative if inside fragment renders nothing', () => {
    const tree = renderer
        .create(<><If condition={false}>{NOT_RENDERED}</If></>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Negative if with evaluated condition', () => {
    let val = 1;
    const tree = renderer
        .create(<div><If condition={val >= 10}>{NOT_RENDERED}</If></div>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Simple if-else', () => {
    const tree = renderer
        .create(
            <div>
                <If condition={false}>{NOT_RENDERED}</If>
                <Else>Condition is false</Else>
            </div>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Nested if', () => {
    const tree = renderer
        .create(
            <If condition={true}>
                <If condition={true}>Both conditions are true</If>
            </If>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Nested if-else', () => {
    const tree = renderer
        .create(
            <If condition={true}>
                <If condition={false}>{NOT_RENDERED}</If>
                <Else><div>1st condition is true but 2nd condition is false</div></Else>
            </If>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Allow whitespace between if-else', () => {
    const tree = renderer
        .create(
            <>
                <If condition={false}>{NOT_RENDERED}</If>


                <Else>Condition was false</Else>
            </>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Else with multiple children', () => {
    const tree = renderer
        .create(
            <>
                <If condition={false}>{NOT_RENDERED}</If>
                <Else>
                    <div>Child 1</div>
                    <div>Child 2</div>
                </Else>
            </>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
