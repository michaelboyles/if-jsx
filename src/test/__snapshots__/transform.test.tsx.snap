// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Allow whitespace between if-else 1`] = `"Condition was false"`;

exports[`Consecutive Ifs 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`Consecutive Ifs with Elses 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`Else preceded by a comment 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`Else where Else is used 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`Else with empty JSX expression 1`] = `null`;

exports[`Else with multiple children 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`ElseIf where only ElseIf passes 1`] = `
Array [
  <div>
    Child 1
  </div>,
  <div>
    Child 2
  </div>,
]
`;

exports[`ElseIf where only If passes 1`] = `
<div>
  Child 1
</div>
`;

exports[`If and ElseIf both pass 1`] = `
<div>
  Child 1
</div>
`;

exports[`If with empty JSX expression 1`] = `null`;

exports[`Multiple ElseIfs 1`] = `
<div>
  Child 1
</div>
`;

exports[`Negative if inside div 1`] = `<div />`;

exports[`Negative if inside fragment renders nothing 1`] = `null`;

exports[`Negative if with evaluated condition 1`] = `<div />`;

exports[`Nested if 1`] = `"Both conditions are true"`;

exports[`Nested if-else 1`] = `
<div>
  1st condition is true but 2nd condition is false
</div>
`;

exports[`Positive if 1`] = `"Condition is true"`;

exports[`Positive if inside div 1`] = `
<div>
  Condition is true
</div>
`;

exports[`Positive if inside fragment 1`] = `"Condition is true"`;

exports[`Positive if with evaluated condition 1`] = `"Condition is true"`;

exports[`Simple if-else 1`] = `
<div>
  Condition is false
</div>
`;
